components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -119,6 +128,54 @@ interface BraveWalletProvider {
   IsLocked() => (bool isLocked);
 };
 
+interface SolanaProvider {
+  // Initializes an SolanaEventsListener
+  Init(pending_remote<SolanaEventsListener> events_listener);
+
+  // Optional {onlyIfTrusted: true}, when that flag is enable, we will only
+  // connect and emit a connect event if the application is trusted.
+  // It will returns base58 encoded public key when success
+  Connect(mojo_base.mojom.DictionaryValue? arg)
+    => (SolanaProviderError error, string error_message, string public_key);
+  Disconnect();
+  [Sync]
+  IsConnected() => (bool is_connected);
+  // return base58 encoded public key
+  [Sync]
```

Comment:
I don't think iOS supports `[Sync]` yet for mojom, not sure if this will compile.

Reply:
Those sync APIs are for `solana.publicKey`, `solana.isConnected`, those property getters have to be sync

Reply:
Well then we have other problems because iOS cannot do synchronous JS communication at all :(

Reply:
you can still access it through async version, `[Sync]` keyword means it would provide addition sync interface for c++

Reply:
https://chromium.googlesource.com/chromium/src/+/HEAD/mojo/public/cpp/bindings/README.md#generated-bindings-c

Reply:
@kylehickinson for `Sync` you just generate a second method signature with a direct return value instead of a callback